presto-memory/src/main/java/io/prestosql/plugin/memory/MemoryMetadata.java

```diff
@@ -294,6 +294,21 @@ else if (views.putIfAbsent(viewName, definition) != null) {
         }
     }
 
+    @Override
+    public synchronized void renameView(ConnectorSession session, SchemaTableName viewName, SchemaTableName newViewName)
+    {
+        checkSchemaExists(newViewName.getSchemaName());
+        if (tableIds.containsKey(newViewName)) {
+            throw new PrestoException(ALREADY_EXISTS, "Table already exists: " + viewName);
+        }
+
+        if (views.containsKey(newViewName)) {
```

Comment:
One of these should be `viewName` and one should be `newViewName`... the exception message is wrong also